sklearn/neighbors/lmnn.py

```diff
@@ -0,0 +1,843 @@
+    that share the same class label. If this is achieved, the leave-one-out
+    error is minimized.
+    This implementation follows closely Kilian Weinberger's MATLAB code found
+    at <https://bitbucket.org/mlcircus/lmnn> which solves the unconstrained
+    problem, finding a linear transformation with L-BFGS instead of solving the
+    constrained problem that finds the globally optimal metric.
+
+
+    Parameters
+    ----------
+    L : array-like
+        Initial transformation in an array with shape (n_features_out,
+        n_features_in).  If None `load` will be used to load a transformation
+        from a file. (default: None)
+
+    n_neighbors : int
+        Number of target neighbors (default: 3)
+
+    max_iter : int
+        Maximum number of iterations in the optimization (default: 200)
+
+    use_pca : bool
+        Whether to use pca to warm-start the linear transformation.
+        If False, the identity will be used. (default: True)
+
+    tol : float
+        Tolerance for the optimization  (default: 1e-5)
+
+    n_features_out : int
+        Preferred dimensionality of the inputs after the transformation.
+        If None it is inferred from `use_pca` and `L`.(default: None)
+
+    max_constr : int
+        Maximum number of constraints to enforce per iteration (default: 10
+        million).
+
+    use_sparse : bool
+        Whether to use a sparse or a dense matrix for the impostor-pairs
+        storage. Using a sparse matrix, the distance to impostors is computed
+        twice, but it is somewhat faster for larger data sets than using a
+        dense matrix. With a dense matrix, the unique impostor pairs have to be
+        identified explicitly (default: True).
+
+    load : string
+        A file path from which to load a linear transformation. If None, either
+        identity or pca will be used based on `use_pca` (default: None).
+
+    save : string
+        A file path prefix to save intermediate linear transformations to.
+        After every function call, it will be extended with the function call
+        number and the `.npy` file extension. If None, nothing will be saved
+        (default: None).
+
+    disp : int, optional
+        If zero, then no output.  If a positive number, then
+        ``0 < disp < 99`` print also f and ``|proj g|`` every disp iterations;
+        ``disp = 99``   print details of every iteration except n-vectors;
+        ``disp = 100``  print also the changes of active set and final x;
+        ``disp > 100``  print details of every iteration including x and g.
+
+    random_state : int
+        A seed for reproducibility of random state  (default: None).
+
+    Attributes
+    ----------
+    L_ : array-like
+        The linear transformation used during fitting with shape
+        (n_features_out, n_features_in).
+
+    n_neighbors_ : int
+        The number of target neighbors (decreased if n_neighbors was not
+        realistic for all classes).
```

Comment:
If so, perhaps "the provided n_neighbors is decreased where >= min(number of elements in each class)"